ls1intum/artemis-android
# `Chore`: Re-update to serialization based navigation

### Problem Description
Tim opened https://github.com/ls1intum/artemis-android/pull/49 that introduced type safe navigation. After prematurely merging the PR, I found that the changes cased a crash. See https://github.com/ls1intum/artemis-android/pull/49#issuecomment-2433172761.

This closes issue #104.

### Changes
 
Readded the changes that Tim already did, and fix two crash causing bugs.


### Steps for testing
Navigate through the app and ensure that everything works as expected. Especially cover the case that the user is not logged in when opening the app.

Cover navigation to the following screens:
- Login
- Registration (eg TS2)
- Dashboard
- ExerciseView
  - Quiz
- LectureScreen
- CommunicationOverview
- Settings
  - PushNotificationsSettings
